File: app/src/main/java/eu/kanade/tachiyomi/ui/catalogue/CatalogueGridHolder.java
package eu.kanade.tachiyomi.ui.catalogue;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.mikepenz.iconics.view.IconicsImageView;

import butterknife.Bind;
import butterknife.ButterKnife;
import eu.kanade.tachiyomi.R;
import eu.kanade.tachiyomi.data.database.models.Manga;

public class CatalogueGridHolder extends CatalogueHolder {

    @Bind(R.id.title) TextView title;
    @Bind(R.id.thumbnail) ImageView thumbnail;
    @Bind(R.id.favorite_sticker) IconicsImageView favoriteSticker;

    public CatalogueGridHolder(View view, CatalogueAdapter adapter, OnListItemClickListener listener) {
        super(view, adapter, listener);
        ButterKnife.bind(this, view);
    }

    @Override
    public void onSetValues(Manga manga, CataloguePresenter presenter) {
        title.setText(manga.title);
        // Set visibility of in library icon.
        favoriteSticker.setVisibility(manga.favorite ? View.VISIBLE : View.GONE);
        // Set alpha of thumbnail.
        thumbnail.setAlpha(manga.favorite ? 0.3f : 1.0f);
        setImage(manga, presenter);
    }

    public void setImage(Manga manga, CataloguePresenter presenter) {
        if (manga.thumbnail_url != null) {
            presenter.coverCache.loadFromNetwork(thumbnail, manga.thumbnail_url,
                    presenter.getSource().getGlideHeaders());
        } else {
            thumbnail.setImageResource(android.R.color.transparent);
        }
    }
}

File: app/src/main/java/eu/kanade/tachiyomi/ui/catalogue/CatalogueGridHolder.kt
package eu.kanade.tachiyomi.ui.catalogue

import android.view.View
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import eu.kanade.tachiyomi.data.database.models.Manga
import kotlinx.android.synthetic.main.item_catalogue_grid.view.*

/**
 * Class used to hold the displayed data of a manga in the catalogue, like the cover or the title.
 * All the elements from the layout file "item_catalogue_grid" are available in this class.
 *
 * @param view the inflated view for this holder.
 * @param adapter the adapter handling this holder.
 * @param listener a listener to react to single tap and long tap events.
 * @constructor creates a new catalogue holder.
 */
class CatalogueGridHolder(private val view: View, private val adapter: CatalogueAdapter, listener: OnListItemClickListener) :
        CatalogueHolder(view, adapter, listener) {

    /**
     * Method called from [CatalogueAdapter.onBindViewHolder]. It updates the data for this
     * holder with the given manga.
     *
     * @param manga the manga to bind.
     */
    override fun onSetValues(manga: Manga) {
        // Set manga title
        view.title.text = manga.title

        // Set alpha of thumbnail.
        view.thumbnail.alpha = if (manga.favorite) 0.3f else 1.0f

        setImage(manga)
    }

    /**
     * Updates the image for this holder. Useful to update the image when the manga is initialized
     * and the url is now known.
     *
     * @param manga the manga to bind.
     */
    fun setImage(manga: Manga) {
        Glide.clear(view.thumbnail)
        if (!manga.thumbnail_url.isNullOrEmpty()) {
            Glide.with(view.context)
                    .load(manga)
                    .diskCacheStrategy(DiskCacheStrategy.SOURCE)
                    .centerCrop()
                    .skipMemoryCache(true)
                    .placeholder(android.R.color.transparent)
                    .into(view.thumbnail)

        }
    }
}